src/ol/renderer/canvas/Layer.js

```diff
@@ -177,6 +177,8 @@ class CanvasLayerRenderer extends LayerRenderer {
       this.container = null;
       this.context = null;
       this.containerReused = false;
+    } else if (this.container && this.container.style) {
```

Comment:
I think this can be simplified to
```suggestion
    } else if (this.container) {
```